### PR TITLE
Modelblocks

### PR DESCRIPTION
**Pull request by rwilliams-msr**
Starting off new space for project around creating block-based models for devices

### CODE REVIEW

**Comment by pelikhan** [comment]
Probably obsolete by https://github.com/microsoft/jacdac-docs/pull/44

File: src/components/model-editor/modelblockdsl.ts [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+export class ModelBlockDomainSpecificLanguage
+    implements BlockDomainSpecificLanguage
+{
+    id = "modelBlocks"
+    createBlocks() {
+        const blocks: BlockDefinition[] = [            
+            /* Dataset blocks */
+            { // would like to make this a hat block
+                kind: "block",
+                type: "dataset_block",
```
Comment (pelikhan) [inline src/components/model-editor/modelblockdsl.ts:21]: maker sure those are globally unique and use macros if you want to reference

File: src/components/model-editor/modelblockdsl.ts [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+import BlockDomainSpecificLanguage from "../blockly/dsl/dsl"
+//import { Block } from "@material-ui/icons"
+import ModelBlockField from "../blockly/fields/ModelBlockField"
+
+const dataset_color = "#123456"
+const class_color = "#2466A8"
+const processing_color = "#ac2469"
+const learning_color = "#561234"
+const layer_color = "#97207a"
+
```
Comment (pelikhan) [inline src/components/model-editor/modelblockdsl.ts:11]: const DATASET_BLOCK = "...."

File: src/components/model-editor/modelblockdsl.ts [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+                type: "class_block",
+                message0: "class %1",
+                args0: [
+                    {
+                      "type": "field_variable", // should be a class type
+                      "name": "CLASS_NAME"
+                    }
+                  ],
+                  inputsInline: false,
+                  previousStatement: null,
```
Comment (pelikhan) [inline src/components/model-editor/modelblockdsl.ts:53]: to restict block that can go use use a type

``
const MODEL_STATEMENT_TYPE = "asidfusoadifu"
...
previousStatement: MODEL_STATEMENT_TYPE 
```

File: src/components/model-editor/modelblockdsl.ts [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+                    "name": "DATASET_CONFIG"
+                  }
+                ],
+                inputsInline: false,
+                colour: dataset_color,
+                tooltip: "Use this block to define your dataset. Create new classes and stack them inside your dataset object",
+                helpUrl: "",
+                hat: true
+              } as BlockDefinition,
+              {
```
Comment (pelikhan) [inline src/components/model-editor/modelblockdsl.ts:42]: or ``<BlockDefinition>{``

**Comment by mmoskal** [comment]
closed in favor #58

**Pull request closed without merging**